core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -35,21 +38,22 @@
 import static org.apache.iceberg.TableProperties.DEFAULT_WRITE_METRICS_MODE_DEFAULT;
 import static org.apache.iceberg.TableProperties.METRICS_MODE_COLUMN_CONF_PREFIX;
 
-public class MetricsConfig implements Serializable {
+@Immutable
+public final class MetricsConfig implements Serializable {
 
   private static final Logger LOG = LoggerFactory.getLogger(MetricsConfig.class);
   private static final Joiner DOT = Joiner.on('.');
 
-  private Map<String, MetricsMode> columnModes = Maps.newHashMap();
-  private MetricsMode defaultMode;
+  private final Map<String, MetricsMode> columnModes;
+  private final MetricsMode defaultMode;
 
-  private MetricsConfig() {
+  private MetricsConfig(Map<String, MetricsMode> columnModes, MetricsMode defaultMode) {
+    this.columnModes = ImmutableMap.copyOf(columnModes);
+    this.defaultMode = defaultMode;
   }
 
   public static MetricsConfig getDefault() {
```

Comment:
Not important for now, but seems like this should return a singleton since these are immutable, so there is only ever one default?